src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -1899,46 +1913,98 @@ internal static void MakeSeparatorListAny(ReadOnlySpan<char> source, ReadOnlySpa
 
         private static void MakeSeparatorListVectorized(ReadOnlySpan<char> sourceSpan, ref ValueListBuilder<int> sepListBuilder, char c, char c2, char c3)
         {
-            // Redundant test so we won't prejit remainder of this method
-            // on platforms where it is not supported
-            if (!Vector128.IsHardwareAccelerated)
+            Debug.Assert(sourceSpan.Length >= Vector128<ushort>.Count);
+            nuint lengthToExamine = (uint)sourceSpan.Length;
+            nuint offset = 0;
+            ref char source = ref MemoryMarshal.GetReference(sourceSpan);
+
+            if (Vector512.IsHardwareAccelerated && lengthToExamine >= (uint)Vector512<ushort>.Count*2)
             {
-                throw new PlatformNotSupportedException();
-            }
```

Comment:
This comment should still be valid. We don't want to be here if V128 isn't supported and want to have it throw.

We can't, today, be in a scenario where V256/V512 are supported but V128 is not.

Reply:
Yeah, that is right. The exception should still be valid. added the code back.